### PR TITLE
dcm2mha preprocessing

### PR DESCRIPTION
**Pull request by joeranbosma**
Implemented:
- Allow user-defined scan postprocessing function
- Made variable names descriptive
- Allow to disable checking DICOM filenames

### CODE REVIEW

File: src/picai_prep/dcm2mha.py [hunk 1]
```diff
@@ -36,6 +37,8 @@ def __init__(
         input_path: PathLike,
         output_path: PathLike,
         settings_path: PathLike,
+        verify_dicom_filenames: bool = True,
+        scan_postprocess_func: Optional[Callable[[sitk.Image], sitk.Image]] = None,
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:41]: Request to minimize the number of complicated non-obvious parameters of the constructor, and instead plop these in a **kwargs. so `kwargs.get('verify_dicom_filenames', True)` instead

Comment (snorthman) [reply src/picai_prep/dcm2mha.py:41]: Furthermore, if put in a `self.kwargs` for example, these can be consumed at an arbitrary point during processing

File: src/picai_prep/dcm2mha.py [hunk 2]
```diff
@@ -187,25 +195,26 @@ def _extract_metadata(self):
                     item['error'] = f'Unexpected error: {e}'
                     self.item_log(item, 'unexpected error')
                 else:
-                    total += len(item['dcms'])
+                    total += (len(item['dcms']) if 'dcms' in item else 0)
 
         self.info(f"Collected {plural(total, 'DICOM file')} from {self.valid_items_str()}.", self.get_history_report())
 
+    @staticmethod
```
Comment (snorthman) [inline src/picai_prep/dcm2mha.py:202]: Why? Only one usage.

**Review by snorthman — CHANGES_REQUESTED** [review]

File: src/picai_prep/dcm2mha.py (hunk 2 above)
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py:202]: Stacked functions make it difficult to follow what's going on, so it there is no benefit, better to keep it simple.

File: src/picai_prep/dcm2mha.py (hunk 1 above)
Comment (joeranbosma) [reply src/picai_prep/dcm2mha.py:41]: Let's tackle this in the Item refactoring?

**Pull request merged**